File: 0283-move-zeroes/0283-move-zeroes.go
func moveZeroes(nums []int)  {
    l, r := 0, 0
    for r < len(nums) {
        if nums[l] == 0 && nums[r] != 0 {
            nums[l], nums[r] = nums[r], nums[l]
        }

        if nums[l] != 0 {
            l++
        }

        r++
    }
}

File: 0283-move-zeroes/0283-move-zeroes.java

class Solution {
    public void moveZeroes(int[] nums) {
        int zeros = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 0) {
                zeros++;
                continue;
            }

            if (zeros > 0) {
                nums[i - zeros] = nums[i];
                nums[i] = 0;
            }
        }
    }
}